### PR TITLE
Add HCA1 registers header

### PR DESCRIPTION
**Pull request by e-puerto**

### CODE REVIEW

File: metal/drivers/sifive_defs.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @file sifive_defs.h
+ * @brief Common define for Sifive drivers
+ */
+
+#pragma GCC diagnostic push
+/* Accept CMSIS syntax */
+#pragma GCC diagnostic ignored "-Wreserved-id-macro"
+
+/* following defines should be used for structure members */
+#define __IM                                                                   \
```
Comment (bsousi5) [inline metal/drivers/sifive_defs.h:17]: Any reason why these need to be in a separate file? If only used for HCA, it should live in hca header file

File: metal/drivers/sifive_hca1_regs.h [hunk 2]
```diff
@@ -0,0 +1,835 @@
...
+        uint32_t MAJORREV:4;       /**< bit:  8..11  HCA Major Revision */
+        uint32_t _reserved0:4;     /**< bit: 12..15  (reserved) */
+        uint32_t CONFIG:5;         /**< bit: 16..20  HCA Configuration */
+        uint32_t _reserved1:11;    /**< bit: 21..31  (reserved) */
+    } b;                           /**< Structure used for bit access */
+    uint32_t w;                    /**< Structure used for word access */
+} HCA_HCA_REV_Type;
+
+/* HCA HCA Revision: Patch Revision */
+#define HCA_HCA_REV_PATCHREV_Pos        0U
```
Comment (bsousi5) [inline metal/drivers/sifive_hca1_regs.h:117]: You should follow the current coding styles, in that all macro/#define should be all caps.
We should also uses the full word and not abbreviation, for POSITION and MASK.

File: metal/drivers/sifive_defs.h (hunk 1 above)
Comment (e-puerto) [reply metal/drivers/sifive_defs.h:17]: I Move it into sifive_hca1_regs.h

File: metal/drivers/sifive_hca1_regs.h (hunk 2 above)
Comment (e-puerto) [reply metal/drivers/sifive_hca1_regs.h]: This file is CMSIS like, so I follow the coding style of CMSIS.

**Pull request merged**